src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -483,6 +483,10 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\PlatformNotSupportedException.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Progress.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Random.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Random.ImplBase.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Random.LegacyImpl.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Random.Xoshiro128StarStarImpl.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Random.Xoshiro256StarStarImpl.cs" />
```

Comment:
You can condition these on `Condition="'$(Platform)' == 'arm64' or '$(Platform)' == 'x64'"` and vice versa so that the assembly only carries one.

Reply:
Ah, right, corelib. Ok.

Reply:
Actually, the build / linker already takes care of that automatically, due to recognition of the IntPtr.Size condition and stripping out the other branch.

Reply:
I prefer to trust what I control myself... !

Reply:
But it would mean,  for example,  the types would need to be named identically.

Reply:
"XoshiroStarStar"

Reply:
And reading the code, what algorithm is that?

Reply:
Presumably you put the names in the comments in the code. We don't require algorithms go in the file name. I guess I like to see what's going on rather than trusting it, but either way is fine.

Reply:
>  I guess I like to see what's going on rather than trusting it

I do, too.  Which is why I'm more confident in what's been used when I see:
```C#
_impl = IntPtr.Size == 8 ? new Xoshiro256() : new new Xoshiro128();
```
rather than:
```C#
_impl = new Xoshiro();
```
and hoping that the build system has been configured correctly to substitute the right implementation based on bitness.

Reply:
ha, fair enough